build.gradle.kts

```
@@ -18,15 +28,16 @@ repositories {
 }
 
 dependencies {
-    api("org.apache.logging.log4j:log4j-core:2.8.1")
-    annotationProcessor("org.apache.logging.log4j:log4j-core:2.8.1")
+    api("org.apache.logging.log4j:log4j-core:2.14.1")
+    annotationProcessor("org.apache.logging.log4j:log4j-core:2.14.1")
 
     api("org.jline:jline-reader:3.20.0")
 
     compileOnly("org.checkerframework:checker-qual:3.13.0")
 
     testImplementation("org.junit.jupiter:junit-jupiter-api:5.7.2")
     testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.7.2")
+    "intTestImplementation"(files(tasks.named<Jar>("jar").get().archiveFile))
```

Sorry, is there a reason you did it like this instead of

```suggestion
    "intTestImplementation"(files(tasks.named("jar")))
```

?

If you do it with the `.get().archiveFile` the implicit dependency on the jar task may be missing, which means the `compileIntTestJava` task could potentially run before the `jar` task and this would fail.

---

I didn't know it was possible to use a task as an input to `files`, so I thought it was an error on your behalf. My mistake; the docs are clear on this.

Interestingly, both the current build file and your suggested change create a dependency on the jar task, according to a [task-info plugin](https://plugins.gradle.org/plugin/org.barfuin.gradle.taskinfo). However, I'll use your approach.

---

> Interestingly, both the current build file and your suggested change create a dependency on the jar task, according to a task-info plugin. However, I'll use your approach.

Yeah I noticed this after some more testing (after I wrote the comment). To be honest, I'm not sure exactly where the dependency is coming from if you do the `.archiveFile`. Very magic :)